prometheus/registry.go

```diff
@@ -722,5 +723,23 @@ func (s metricSorter) Less(i, j int) bool {
 			return vi < vj
 		}
 	}
-	return true
+	if s[i].GetTimestampMs() != s[j].GetTimestampMs() {
+		// Ensure metrics are ordered by timestamp.  Treat
+		// a zero timestamp as "now" in case we have
+		// untimestamped metrics mixed with timestamped
+		// metrics.
+		ti, tj := s[i].GetTimestampMs(), s[j].GetTimestampMs()
+		if ti == 0 {
```

Comment:
Sounds good.  I've updated to order a nil timestamp at the end (most likely that is the best "now" location) and eliminated the dependency on time.  PTAL?